File: cpp/alerts/storage_moved_failed.cpp
#include "./storage_moved_failed.h"

namespace ltrs {

rust::Str storage_moved_failed_alert_file_path(lt::storage_moved_failed_alert* alert) {
    return alert->file_path();
}

Error storage_moved_failed_alert_get_error(lt::storage_moved_failed_alert* alert) {
    return error_code_to_error(alert->error);
}

uint8_t storage_moved_failed_alert_get_op(lt::storage_moved_failed_alert* alert) {
    return static_cast<uint8_t>(alert->op);
}

} // namespace ltrs

File: src/ffi/alerts/storage_moved_failed.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type storage_moved_failed_alert = crate::ffi::ffi::storage_moved_failed_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/storage_moved_failed.h");
        type Error = crate::ffi::error::ffi::Error;
        unsafe fn storage_moved_failed_alert_file_path<'a>(
            alert: *mut storage_moved_failed_alert,
        ) -> &'a str;
        unsafe fn storage_moved_failed_alert_get_error(
            alert: *mut storage_moved_failed_alert,
        ) -> Error;
        unsafe fn storage_moved_failed_alert_get_op(alert: *mut storage_moved_failed_alert) -> u8;
    }
}
